docs/reference/buildx_build.md

```diff
@@ -126,6 +126,30 @@ FROM alpine
 COPY --from=project myfile /
 ```
 
+Source an image from a local [OCI layout compliant directory](https://github.com/opencontainers/image-spec/blob/main/image-layout.md):
+
+```console
+$ docker buildx build --build-context foo=oci-layout:///path/to/local/layout@sha256:abcd12345 .
```

Comment:
I think we should allow this input without the digest as well in here. If no digest is set it can look up `index.json` inside this dir and see if it has one descriptor and then set it as the expected digest. If there is ambiguity then it can produce an error.

Reply:
I am hesitant about this, as it wades into a deep unresolved issue around what actually should be in `index.json`. I have seen those that _only_ include tags, others that include entire image refs, making the layout directory essentially into a full registry. I have implemented. both ways.

containerd does a bit of a hybrid: directory _is_ a full registry, but tag->manifest-hash is in the boltdb.

Let's get this working with the hash required, as is, and then let's deal with supporting hash-less - both in buildkit and in buildx - as a separate iteration.